text_extensions_for_pandas/array/token_span.py

```diff
@@ -483,14 +494,18 @@ def _from_sequence(cls, scalars, dtype=None, copy=False):
         i = 0
         for s in scalars:
             if not isinstance(s, TokenSpan):
-                raise ValueError(
-                    f"Can only convert a sequence of TokenSpan "
-                    f"objects to a TokenSpanArray. Found an "
-                    f"object of type {type(s)}"
-                )
-            if tokens is None:
+                # TODO: Temporary fix for np.nan values, pandas-dev GH#38980
+                if np.isnan(s):
+                    s = TokenSpanDtype().na_value
+                else:
+                    raise ValueError(
+                        f"Can only convert a sequence of TokenSpan "
+                        f"objects to a TokenSpanArray. Found an "
+                        f"object of type {type(s)}"
+                    )
+            if tokens is None and not (s.begin == TokenSpan.NULL_OFFSET_VALUE and s.tokens.target_text is None):
```

Comment:
This gets a little awkward dealing with with a NULL value with target_text=None, but should be ok for now